test/integration/podgc/podgc_test.go

```diff
@@ -74,76 +74,79 @@ func TestPodGcOrphanedPodsWithFinalizer(t *testing.T) {
 	}
 
 	for name, test := range tests {
-		t.Run(name, func(t *testing.T) {
-			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodDisruptionConditions, test.enablePodDisruptionConditions)()
-			testCtx := setup(t, "podgc-orphaned")
-			cs := testCtx.ClientSet
+		for _, podReplacementPolicyEnable := range []bool{false, true} {
```

Comment:
I would suggest introducing enablePodReplacementPolicy as a testcase parameter. It does not seem obvious why we need to multiply all test cases by this feature. Then, only duplicate the test cases where the feature really makes a difference.

Reply:
Some combinations are also not obvious, for example, when we have `"PodDisruptionConditions disabled"`, then when  `podReplacementPolicyEnable: true` I would expect the `wantPhase` to be `Failed`.